src/Kernel32.Desktop/Kernel32.Helpers.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    /// <content>
+    /// Methods and nested types that are not strictly P/Invokes but provide
+    /// a slightly higher level of functionality to ease calling into native code.
+    /// </content>
+    public static partial class Kernel32
+    {
+        public static PROCESSENTRY32 Process32First(SafeObjectHandle hSnapshot)
+        {
+            if (hSnapshot == null)
+            {
+                throw new ArgumentNullException(nameof(hSnapshot));
+            }
+
+            var entry = new PROCESSENTRY32();
+            if (Process32First(hSnapshot, entry))
+            {
+                return entry;
+            }
+
+            var lastError = (Win32ErrorCode)Marshal.GetLastWin32Error();
+            if (lastError != Win32ErrorCode.ERROR_NO_MORE_FILES)
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
+            }
+
+            return null;
+        }
+
+        public static PROCESSENTRY32 Process32Next(SafeObjectHandle hSnapshot)
+        {
+            if (hSnapshot == null)
+            {
+                throw new ArgumentNullException(nameof(hSnapshot));
+            }
+
+            var entry = new PROCESSENTRY32();
+            if (Process32Next(hSnapshot, entry))
+            {
+                return entry;
+            }
+
+            var lastError = (Win32ErrorCode)Marshal.GetLastWin32Error();
+            if (lastError != Win32ErrorCode.ERROR_NO_MORE_FILES)
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
```

Comment:
Instead of this can you call lastError.ThrowOnError()? Now that `Win32Exception` works, lets throw that instead of `COMException`.